python/hsml/utils/columnar_schema.py

```diff
@@ -0,0 +1,107 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from hsml.utils.column import Column
+import pandas
+import importlib
+
+try:
+    import hsfs
+except ImportError:
+    pass
+
+try:
+    import pyspark
+except ImportError:
+    pass
+
+
+class ColumnarSchema:
+    """Metadata object representing a columnar schema for a model."""
+
+    def __init__(self, columnar_obj=None):
+        if isinstance(columnar_obj, list):
+            self.columns = self._convert_list_to_schema(columnar_obj)
+        elif isinstance(columnar_obj, pandas.DataFrame):
+            self.columns = self._convert_pandas_df_to_schema(columnar_obj)
+        elif isinstance(columnar_obj, pandas.Series):
+            self.columns = self._convert_pandas_series_to_schema(columnar_obj)
+        elif importlib.util.find_spec("pyspark") is not None and isinstance(
+            columnar_obj, pyspark.sql.dataframe.DataFrame
+        ):
+            self.columns = self._convert_spark_to_schema(columnar_obj)
+        elif importlib.util.find_spec("hsfs") is not None and isinstance(
+            columnar_obj, hsfs.training_dataset.TrainingDataset
+        ):
+            self.columns = self._convert_td_to_schema(columnar_obj)
+        else:
+            raise TypeError(
+                "{} is not supported in a schema.".format(type(columnar_obj))
```

Comment:
```suggestion
                "{} is not supported in a columnar schema.".format(type(columnar_obj))
```
Same comment as for tensor schema.